.github/workflows/phpunit-tests.yml

```diff
@@ -47,64 +38,11 @@ jobs:
         with:
           access_token: ${{ github.token }}
 
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Log debug information
-        run: |
-          echo "$GITHUB_REF"
-          echo "$GITHUB_EVENT_NAME"
-          npm --version
-          node --version
-          curl --version
-          git --version
-          svn --version
-          php --version
-          php -i
-          locale -a
-
-      - name: Install NodeJS
-        uses: actions/setup-node@v2
-        with:
-          node-version: 14
-
-      - name: Cache NodeJS modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
-        with:
-          # npm cache files are stored in `~/.npm` on Linux/macOS
-          path: ~/.npm
-          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-npm-
-
-      - name: Install Dependencies
-        run: npx install-changed --install-command="npm ci"
-
-      - name: Build WordPress
-        run: npm run build
```

Comment:
Yes I second this. We should generalize that workflow and test on more OSes if we're not building in the PHPUnit test workflow.

Reply:
I forgot to mention in the PR description that `npm run build` still gets run as part of the QUnit tests, so if the new build assertions in Gruntfile.js don't pass then the JavaScript test workflow is the one that fails. I agree that it's worth an additional explicit workflow for MacOS though. Not sure if there's need for another separate one for Linux? Maybe.

Reply:
I don't think it would hurt to have a separate job that explicitly shows the build task runs successfully. I'm going to open a new ticket for this, though.

Reply:
This is where cross-workflow dependencies in GHA would be useful. The JS workflow could depend on this, so we don't run the build twice.

Reply:
https://core.trac.wordpress.org/ticket/52658